app/renderer/bookmarks.js

```diff
@@ -0,0 +1,157 @@
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+"use strict"
+const {
+    writeJSON,
+    readJSON,
+    joinPath,
+    notify,
+    isAbsolutePath,
+    appData
+} = require("../util")
+const {
+    currentPage,
+    getSetting,
+    currentTab
+} = require("./common")
+
+let bookmarkData = {}
+let bookmarksFile = ""
+
+const validOptions = [
+    "bg",
+    "fg",
+    "keywords",
+    "name",
+    "path",
+    "tag",
+    "title",
+    "url"
+]
+
+const setBookmarkSettings = () => {
+    const setFile = getSetting("bookmarksfile")
+    if (setFile === "bookmarks" || setFile.trim() === "") {
+        bookmarksFile = joinPath(appData(), "bookmarks")
+    } else if (isAbsolutePath(setFile)) {
+        bookmarksFile = setFile
+    } else {
+        bookmarksFile = joinPath(appData(), setFile)
+    }
+    bookmarkData = readJSON(bookmarksFile)
+}
+
+const addBookmark = input => {
+    const newbookmark = {}
+    // If there's "=" or "~" we parse the input.
+    if (input.join().includes("=") || input.join().includes("~")) {
+        // Retain spaces for multiple word inputs.
+        const options = input.join(" ").split("~")
+        for (let i = 0; i < options.length; i++) {
+            // Get key and value: [0,1]
+            const keyAndValue = options[i].split("=")
+            if (keyAndValue[0] === "keywords" || keyAndValue[0] === "tag") {
+                const tagsOrKeywordList = keyAndValue.slice(1).join("")
+                    .split(",")
+                if (tagsOrKeywordList.length > 0) {
+                    newbookmark[keyAndValue[0]] = tagsOrKeywordList
+                }
+            } else {
+                const allValue = keyAndValue.slice(1).join("")
+                if (allValue?.trim()) {
+                    newbookmark[keyAndValue[0]] = allValue
+                }
+            }
+        }
+    } else {
+        // If there's only text, we take it as name for current page bookmark
+        newbookmark.name = input.join(" ")
+        if (newbookmark.name?.trim()) {
+            // When user doesn't enter anything delete empty string
+            delete newbookmark.name
+        }
+    }
+    // Fill missing essential data from relevant sources
+    if (!newbookmark.name?.trim()) {
+        newbookmark.name = currentTab().querySelector("span").textContent
+    }
+    if (!newbookmark.title?.trim()) {
+        newbookmark.title = currentTab().querySelector("span").textContent
+    }
+    if (!newbookmark.url?.trim()) {
+        newbookmark.url = currentPage().src
+    }
+    if (!newbookmark.path?.trim()) {
+        newbookmark.path = "/"
+    }
+    if (!newbookmark.bg?.trim()) {
+        newbookmark.bg = "some default hex value"
```

Comment:
You can use the theme colors as the defaults, for example:
https://github.com/Jelmerro/Vieb/blob/master/app/renderer/settings.js#L958

Reply:
Ok, but now thinking about it, if someone changes their theme, they're going to end up with some hard coded themed values in every bookmark. 
Those could be changed later, but i don't know, it seems bad to me. 
Maybe we could make it so this option accepts something like bg-default and fg-default, that then resolve to the defaults of the theme while rendering or something like that. What do you think?

Reply:
That's a valid concern actually, maybe we should not add the colors to every entry, and just fallback to the theme colors in case they are missing when it actually happens, not when making them.

Reply:
For now i've just removed the auto filling of those values.